content/blog/wireshark-protobuf-and-grpc-dissectors.md

```diff
@@ -0,0 +1,243 @@
+---
+title: Wireshark Protobuf and gRPC dissectors
+date: 2020-10-20
+spelling: cSpell:ignore Huang Qiangxiong
+author:
+  name: Huang Qiangxiong
+---
+
+[Wireshark](https://www.wireshark.org/) is the most famous open source network protocol analyzer, which is widely used in network troubleshooting, analysis, software and communications protocol development, and education. Here is the history of Wireshark Protobuf and gRPC dissectors:
+
+- Wireshark 2.6.0 - adds the initial version of Protobuf and gRPC dissectors.
+
+- Wireshark 3.2.0 - supports dissecting serialized Protobuf data (such as gRPC) more precisely according to *.proto files, and supports dissecting gRPC messages of streaming calls.
+
+- Wireshark 3.3.0 - fixes some bugs about parsing *.proto files and supports some new features about Protobuf dissector.
+
+## Features
+
+The main features about Wireshark Protobuf and gRPC dissectors includes:
+
+- Supports dissecting gRPC messages serialized in Protobuf or JSON format.
+
+- Supports dissecting the gRPC messages of unary, server streaming, client streaming, and bidirectional streaming RPC calls.
+
+- Message and service definition language files (*.proto) can be configured to enable more precise parsing of the serialized data in Protobuf format for protocols such as gRPC.
+
+- Protobuf fields can be dissected as Wireshark fields that allows user to input the full name of a Protobuf field or message in the filter toolbar to search packages containing the field or message.
+
+- You can register your own subdissectors in the 'protobuf_field' dissector table, which is keyed with the full names of the fields, for further parsing Protobuf fields of BYTES or STRING type.
+
+- You can add the mapping record between UDP port and Protobuf message type in the Protobuf protocol preferences dialog, so that the Wireshark can decode the traffic on that UDP port as this Protobuf message.
+
+- You can create your own dissectors to parsing the protocols based on Protobuf over UDP or TCP.
+
+## Examples about using Protobuf and gRPC dissectors
+
+There are some examples to show how to use the Wireshark Protobuf and gRPC dissectors. You can get more details about these examples from [Protobuf wiki](https://gitlab.com/wireshark/wireshark/-/wikis/Protobuf) and [gRPC wiki](https://gitlab.com/wireshark/wireshark/-/wikis/gRPC) pages from Wireshark official website.
```

Comment:
Probably want to include "page" within the link itself; as-is it is very easy to think those are protobuf and grpc wikis as opposed to pages on the wireshark wiki.

Reply:
done